### DISCUSSION TITLE
calculate time and distance

### DISCUSSION THREAD

**Discussion by ruqaiyasattar** [fleaflet/flutter_map#1739] [Q&A] [answered]
how to Calculate time and distance from one location to another? I did get distance using Distance class

final Distance distance = const Distance();
 final double km = distance.as(LengthUnit.Kilometer, _jeddah ,_riyad );

But need guidance for time calculation

**Comment by JaffaKetchup** [comment, marked as answer]
Note that that will give you the straight line distance.

To find the driving distance and time between two points, you'll need a routing machine/API, for example the Google Maps Platform, Mapbox Routing, or a more open source option like OSRM or Valhalla. More information can be found on our docs site.

**Comment by ruqaiyasattar** [comment]
Do I need to use other Open source solution than flutter_map?

**Reply by JaffaKetchup** [reply]
Yes, routing is out of scope for us. This is clearly explained in documentation.
